omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -587,10 +582,27 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
                     self._table.numberOfPages = Math.floor(initialRowCount/self._table.fnSettings()._iDisplayLength)+1;
 
             } else if(self.options.searchPhrase || self.options.doSearchWhenEmpty) {
-                if (self.options.searchPhrase == null) {
-                    self.options.searchPhrase = "";
-                }
-                $j(input).val(self.options.searchPhrase).keyup();
+            	if (self.options.searchPhrase == null) {
+            		self.options.searchPhrase = "";
+            	}                
+            	//check to see if we need to select the checkbox for includeVerbose and/or includeVoided
+            	//since there is a searchPhrase the page will reload using these values
+            	if(self.options.showVerbose){
+            		$j('#includeVerbose').attr('checked','checked');
+            	}
+            	if(self.options.includeVoided){
+            		$j('#includeVoided').attr('checked','checked');
+            	}
+            	$j(input).val(self.options.searchPhrase).keyup();
+            	//setting the cursor to the end of the searchPhrase
+            	//setSelectionRange is a method of the DOM element not the jQuery object so get the DOM element
+            	var inputNodeDOM = $j('#inputNode').get(0);
+            	if(inputNodeDOM){
+            		var searchPhraseLength = self.options.searchPhrase.length;
+            		if (inputNodeDOM.setSelectionRange) {
+            			inputNodeDOM.setSelectionRange(searchPhraseLength, searchPhraseLength);
```

Comment:
What the logic seems to be doing is to select(highlight) the text set in the search input box although the comment says it's placing the cursor to the end, i personally see no benefit to it and i suspect it can fail in some browsers. Can you test it out in the browser to see the behavior?